### PR TITLE
Check for obvious contradicting assumptions. 

### PR DESCRIPTION
**Pull request by wujingyue**
Helps #1572

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

File: csrc/expr_simplifier.cpp [hunk 1]
```diff
@@ -188,6 +189,26 @@ class Context {
         assumptions.push_back(bop->rhs());
       }
     }
+
+    // Check for obvious contradictions. Could have helped debugging #1572.
+    // Let me know if there's a subquadratic way to check this. `Val*` equality
+    // was't sufficient, so I had to call `sameAs` instead.
+    for (const auto& [a, b] : getKnownLessThan()) {
+      for (const auto& [x, y] : getKnownLessEqual()) {
+        // a < b && b <= a is impossible.
```
Comment (jacobhinkle) [inline csrc/expr_simplifier.cpp:198]: We could add an inner loop to check `a < b && b < a`.

File: csrc/expr_simplifier.cpp [hunk 2]
```diff
@@ -23,6 +23,7 @@
 #include <memory>
 #include <numeric>
 #include <regex>
+#include <set>
```
Comment (jacobhinkle) [inline csrc/expr_simplifier.cpp:26]: Is this needed?

**Review by jacobhinkle — COMMENTED** [review]
Makes sense to me, but I'll let @zasdfgbnm stamp. Note we may want to factor this into a `bool isConsistent() const` method.

**Comment by xwang233** [comment]
!build

**Review by zasdfgbnm — COMMENTED** [review]
Thanks for fixing this! Other than Jacob's comment, this PR makes sense to me. Will let Jacob stamp.

**Comment by wujingyue** [comment]
!build

File: csrc/expr_simplifier.cpp (hunk 1 above)
Comment (wujingyue) [reply csrc/expr_simplifier.cpp:217]: Done.

File: csrc/expr_simplifier.cpp (hunk 2 above)
Comment (wujingyue) [reply csrc/expr_simplifier.cpp:26]: Removed.

**Review by jacobhinkle — APPROVED** [review]
LGTM!

File: csrc/expr_simplifier.cpp [hunk 3]
```diff
@@ -188,6 +188,45 @@ class Context {
         assumptions.push_back(bop->rhs());
       }
     }
+
+    validateConsistency();
+  }
+
+  void validateConsistency() {
```
Comment (jacobhinkle) [inline csrc/expr_simplifier.cpp:195]: Noticed a nit too late: `const` :-D.

**Pull request merged**